docs/ref/models/database-functions.txt

```diff
@@ -293,3 +293,518 @@ Usage example::
+    ...    start_time__day=ExtractDay('start_time')).count()
+    1
+
+``ExtractWeekDay``
+------------------
+
+.. class:: ExtractWeekDay(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'week_day'
+
+Logically equivalent to ``Extract('date_field', 'week_day')``.
+``ExtractWeekDay`` is also a ``Transform`` registered on Date and DateTime
+fields as ``__week_day``. Returns the week_day of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractWeekDay
+    >>> start = datetime(2015, 6, 15)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_week_day=ExtractWeekDay('start_time')).get()
+    >>> print(experiment.start_week_day)
+    2
+    >>> Experiment.objects.filter(
+    ...    start_time__week_day=ExtractWeekDay('start_time')).count()
+    1
+
+``ExtractHour``
+---------------
+
+.. class:: ExtractHour(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'hour'
+
+Logically equivalent to ``Extract('date_field', 'hour')``. ``ExtractHour`` is
+also a ``Transform`` registered on Date and DateTime fields as ``__hour``.
+Returns the hour of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractHour
+    >>> start = datetime(2015, 6, 15, 14)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_hour=ExtractHour('start_time')).get()
+    >>> print(experiment.start_hour)
+    14
+    >>> Experiment.objects.filter(
+    ...    start_time__hour=ExtractHour('start_time')).count()
+    1
+
+``ExtractMinute``
+-----------------
+
+.. class:: ExtractMinute(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'minute'
+
+Logically equivalent to ``Extract('date_field', 'minute')``. ``ExtractMinute``
+is also a ``Transform`` registered on Date and DateTime fields as ``__minute``.
+Returns the minute of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
+    >>> from django.db.models.functions import ExtractMinute
+    >>> start = datetime(2015, 6, 15, 14, 30)
+    >>> Experiment.objects.create(start_time=start)
+    >>> experiment = Experiment.objects.annotate(
+    ...    start_minute=ExtractMinute('start_time')).get()
+    >>> print(experiment.start_minute)
+    30
+    >>> Experiment.objects.filter(
+    ...    start_time__minute=ExtractMinute('start_time')).count()
+    1
+
+``ExtractSecond``
+-----------------
+
+.. class:: ExtractSecond(date_expression, **extra)
+
+    .. attribute:: lookup_name = 'second'
+
+Logically equivalent to ``Extract('date_field', 'second')``. ``ExtractSecond``
+is also a ``Transform`` registered on Date and DateTime fields as ``__second``.
+Returns the second of ``date_expression``.
+
+Usage example::
+
+    >>> from datetime import datetime
```

Comment:
Yes, I think the documentation would be more clear if you factored:

- extracts that operate on dates and datetimes (year / month / day / week_day)
- extracts that operate only on datetimes (hour / minute / second)